packages/curriculum-parser-markdown/plugins/insight/yaml.js

```diff
@@ -76,12 +76,14 @@ function getMarkdownLink(link) {
 
 // http://stackoverflow.com/questions/8498592/extract-root-domain-name-from-string
 function getDomainFromURL(url) {
-  return url
+  const urlDomainArray = url
     ? // find & remove protocol (http, ftp, etc.) and get domain
       (url.indexOf('://') > -1 ? url.split('/')[2] : url.split('/'))
         // find & remove port number
         .split(':')
     : null;
```

Comment:
Then you should probably make this line `[]` so that array access on line 86 below doesn't cause an exception:

```suggestion
    : [null];
```

Reply:
good shout :+1:

Reply:
Although this whole function could now be handled by the native url:

```js
if (!url) return null;
return new URL(d).hostname;
```

Reply:
can do :+1: I didn't know `URL` can do this

Reply:
hmm, URL is supported from node 10, but the package uses node 8 or higher

```
The 'URL' is not supported until Node.js 10.0.0. The configured version range is '>=8.0.0'.
```

should we update the node version?